src/mongocxx/collection.hpp

```diff
@@ -47,6 +47,8 @@
 #include <mongocxx/result/update.hpp>
 #include <mongocxx/write_concern.hpp>
 
+#include <mongoc.h>
```

Comment:
Hmm you're right it's probably not ok.  I will think about what to do about that constructor...